skbio/sequence/_sequence.py

```diff
@@ -223,13 +464,19 @@ def __init__(self, sequence, metadata=None,
             if positional_metadata is None:
                 positional_metadata = sequence.positional_metadata
             sequence = sequence._bytes
-        else:
-            if metadata is None:
-                metadata = {}
 
         self._set_sequence(sequence)
-        self.metadata = metadata
-        self._set_positional_metadata(positional_metadata)
+
+        if metadata is None:
+            # reset to default empty metadata representation
+            del self.metadata
```

Comment:
I know this is being optimistic, but you could also do `self._metadata = None` and modify the getter slightly